test/currencyConvertor.ts

```diff
@@ -1,73 +1,196 @@
+    // approve ERC20 contracts
+    await usdtTokenContract.approve(
+      currencyConvertor.address,
+      100000000000,
+    );
   });
+
+  after(async () => {
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [impersonatedAccount],
+    });
+  })
+
   describe("deposit", async () => {
-    it("deposit USDT to USDC", async () => {
+    it("deposit USDT as USDC to Starkware", async () => {
+      const zeroExTransaction = await zeroExRequest('100');
+
+      // get old balances
+      const userUsdtBalance: BigNumber = await usdtTokenContract.balanceOf(signer.address);
+      const starkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      )
+
+      await zeroExExchangeWrapper.approveSwap(zeroExTransaction.to, usdtTokenAddress);
+      const tx = await currencyConvertor.deposit(
+        usdtTokenAddress,
+        '100000',
+        zeroExExchangeWrapper.address,
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
+        ethers.utils.getAddress(zeroExTransaction.to),
+        zeroExTransaction.data,
+      );
+
+      const blocks = await tx.wait();
+      const events = _.chain(blocks.events!)
+      .filter((e) => e.event === 'LogConvertedDeposit')
+      .value();
+
+      const event = events[0];
+      expect(event.args?.tokenFromAmount.toString()).to.equal('100000');
+      expect(event.args?.tokenFrom.toLowerCase()).to.equal(usdtTokenAddress);
+
+      // get new balances
+      const newUserUsdtBalance: BigNumber = await usdtTokenContract.balanceOf(signer.address);
+      const newStarkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      )
+
+      expect(newUserUsdtBalance.lt(userUsdtBalance)).to.be.true;
+      expect(newStarkwareUsdcBalance.gt(starkwareUsdcBalance)).to.be.true;
+
+      // deposit with approvals
+      const zeroExTransaction2 = await zeroExRequest('100');
       await currencyConvertor.deposit(
-        usdc.address,
-        100,
+        usdtTokenAddress,
+        '100000',
         zeroExExchangeWrapper.address,
-        100,
-        1,
-        0x4,
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
+        ethers.utils.getAddress(zeroExTransaction.to),
+        zeroExTransaction2.data,
       );
     });
+
+    it("deposit USDT to USDC without enough funds", async () => {
+      const zeroExTransaction = await zeroExRequest('1000000');
+
+      await zeroExExchangeWrapper.approveSwap(zeroExTransaction.to, usdtTokenAddress);
+      await expect(currencyConvertor.deposit(
+        usdtTokenAddress,
+        '1',
+        zeroExExchangeWrapper.address,
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
```

Comment:
Can you add tests where the Starkware contract throws an error (such as not having a registered STARK key)?